packages/design-system/src/components/Switch/Switch.tsx

```diff
@@ -1,94 +1,111 @@
 import * as React from 'react';
 
+import * as Switch from '@radix-ui/react-switch';
+import { composeEventHandlers } from '@strapi/ui-primitives';
 import { styled } from 'styled-components';
 
-import { Box } from '../Box';
+import { useControllableState } from '../../hooks/useControllableState';
 import { Flex } from '../Flex';
+import { Typography, TypographyComponent } from '../Typography';
 
-interface SwitchProps extends Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'onChange'> {
-  label: string;
-  onChange: React.MouseEventHandler<HTMLButtonElement>;
+interface SwitchProps extends Omit<Switch.SwitchProps, 'children'> {
   onLabel?: string;
   offLabel?: string;
-  selected?: boolean;
   visibleLabels?: boolean;
 }
 
-const Switch = React.forwardRef<HTMLButtonElement, SwitchProps>(
-  ({ label, onChange, onLabel = 'On', offLabel = 'Off', selected, visibleLabels = false, ...props }, ref) => {
+const SwitchImpl = React.forwardRef<HTMLButtonElement, SwitchProps>(
+  (
+    {
+      visibleLabels,
+      onLabel = 'On',
+      offLabel = 'Off',
+      onCheckedChange: onCheckedChangeProp,
+      checked: checkedProp,
```

Comment:
I think instead of renaming props, we could just alias as you already did, to avoid making changes everywhere its used.

```suggestion
      onChange: onCheckedChangeProp,
      selected: checkedProp,
```

Reply:
ohk, but in this context `onChange` and `onCheckedChange` wont be same? what could be the other event

Reply:
onCheckedChange gets `checked` state so it's recieves a boolean value, onChange is the event, but if it's controlled it won't have changed iirc. Most of our usecases, we should prefer `onCheckedChange` it's a lot easier to work with. The checkbox component also has the state `indeterminate` which therefore avoids the need for another prop 🤓

Reply:
thanks for the explanation!! 👍